### PR TITLE
feat: Configurable Accounting Dimensions

### PR DESCRIPTION
**Pull request by deepeshgarg007**
Docs PR:https://github.com/frappe/erpnext_com/pull/354

Creating Dimensions:

accounting-dimension

Accounting Dimension Section:

dimension-section

Filters in General Ledger:

report-dimension

### CODE REVIEW

File: erpnext/accounts/doctype/accounting_dimension/accounting_dimension.py [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+		if frappe.flags.in_test:
+			make_dimension_in_accounting_doctypes(doc=self)
+		else:
+			frappe.enqueue(make_dimension_in_accounting_doctypes, doc=self)
+
+	def on_trash(self):
+		if frappe.flags.in_test:
+			delete_accounting_dimension(doc=self)
+		else:
+			frappe.enqueue(delete_accounting_dimension, doc=self)
```
Comment (rmehta) [inline erpnext/accounts/doctype/accounting_dimension/accounting_dimension.py]: show progress

File: erpnext/accounts/doctype/accounting_dimension/accounting_dimension.py [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+		df = {
+			"fieldname": doc.fieldname,
+			"label": doc.label,
+			"fieldtype": "Link",
+			"options": doc.document_type,
+			"insert_after": insert_after_field
+		}
+
+		if doctype == "Budget":
+			df.update({
```
Comment (rmehta) [inline erpnext/accounts/doctype/accounting_dimension/accounting_dimension.py]: break into smaller function `add_to_budget`

File: erpnext/accounts/doctype/accounting_dimension/accounting_dimension.py [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+	for doctype in doclist:
+		field = frappe.db.get_value("Custom Field", {"dt": doctype, "fieldname": doc.get('fieldname')})
+		if field:
+			custom_field = frappe.get_doc("Custom Field", field)
+			custom_field.update(df)
+			custom_field.save()
+
+		frappe.clear_cache(doctype=doctype)
+
+def get_doclist():
```
Comment (rmehta) [inline erpnext/accounts/doctype/accounting_dimension/accounting_dimension.py]: `get_doctypes_with_dimensions`

**Review by rmehta — CHANGES_REQUESTED** [review]

**Review by rmehta — APPROVED** [review]

**Comment by jaichavan** [comment]
Awesome!!!

**Comment by Muhammadinaam** [comment]
Very nice feature. I have a small question. If we use the dimension feature for different branches of a company then what will be the effect on inventory valuation. We want the inventory valuation (Perpetual FIFO of Moving Average) of each branch to be separate. Is it possible with the current feature?

**Pull request merged**